<a href="https://colab.research.google.com/github/GibsonKingori/-AI-Coursework2025-/blob/main/NetflixTitles-DataCleaning.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import files
uploaded = files.upload()
import pandas as pd



Saving netflix_titles.csv to netflix_titles (2).csv


# New Section

In [ ]:
df = pd.read_csv('netflix_titles.csv')

df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8807 entries, 0 to 8806
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype 
---  ------        --------------  ----- 
 0   show_id       8807 non-null   object
 1   type          8807 non-null   object
 2   title         8807 non-null   object
 3   director      6173 non-null   object
 4   cast          7982 non-null   object
 5   country       7976 non-null   object
 6   date_added    8797 non-null   object
 7   release_year  8807 non-null   int64 
 8   rating        8803 non-null   object
 9   duration      8804 non-null   object
 10  listed_in     8807 non-null   object
 11  description   8807 non-null   object
dtypes: int64(1), object(11)
memory usage: 825.8+ KB


Question 1
How many rows and columns are in this dataset

In [ ]:
df.shape


(8807, 12)

***There are 8807 rows and 12 columns in this dataset ***





Question 2
What datatypes are conatined in the dataset ?

In [ ]:
df.dtypes

show_id         object
type            object
title           object
director        object
cast            object
country         object
date_added      object
release_year     int64
rating          object
duration        object
listed_in       object
description     object
dtype: object

Answer : Most columns are object and Int


Question 3

Is the  dataset complete ? ie no missing gvalues

In [ ]:
df.isnull().sum()

show_id            0
type               0
title              0
director        2634
cast             825
country          831
date_added        10
release_year       0
rating             4
duration           3
listed_in          0
description        0
dtype: int64

The data set is not complete  some columns  such as director ,cast ,country ,date_added,rating ,duration are incomplete and have some  missing values

Question 4
List the number of movies and TV shows amont the last 20 rows of the dataset

In [ ]:
df.tail(20)['type'].value_counts()


type
Movie      15
TV Show     5
Name: count, dtype: int64

In the last 20 rows there are 15 movies and 5 TV shows .

Question 5
Give the number of all movies originatting from South Africa .

In [ ]:
sa_movies = df[(df['type'] == 'Movie') & (df['country'] == 'South Africa')]
len(sa_movies)


24

There are 24 Movies originating from South Africa in this Dataset

Question 6
What is the description of the movie in the 12th row

In [ ]:
df.iloc[11]['description']


'Struggling to earn a living in Bangkok, a man joins an emergency rescue service and realizes he must unravel a citywide conspiracy.'

The description is
"Struggling to earn a living in Bangkok, a man joins an emergency rescue service and realizes he must unravel a citywide conspiracy.""

Question 7
From this South African Movies list only those released in 2021

In [ ]:
sa_2021 = sa_movies[sa_movies['release_year'] == 2021]
sa_2021[['title', 'release_year']]


title  release_year
293                           Slay          2021
764   Trippin' with the Kandasamys          2021
881                 I Am All Girls          2021
1172       Loyiso Gola: Unlearning          2021

From the dataset there were 4 south African movies released in 2021

Question 8

List the  enitre cast of all movies released in 2021

In [ ]:
cast_2021 = df[(df['type'] == 'Movie') & (df['release_year'] == 2021)][['title', 'cast']]
cast_2021


title  \
6                      My Little Pony: A New Generation   
9                                          The Starling   
12                                         Je Suis Karl   
13                     Confessions of an Invisible Girl   
18                                            Intrusion   
...                                                 ...   
1425                        Creating The Queen's Gambit   
1431                                        Stuck Apart   
1436  Ratones Paranoicos: The Band that Rocked Argen...   
1466                       The Minimalists: Less Is Now   
1468                          What Happened to Mr. Cha?   

                                                   cast  
6     Vanessa Hudgens, Kimiko Glenn, James Marsden, ...  
9     Melissa McCarthy, Chris O'Dowd, Kevin Kline, T...  
12    Luna Wedler, Jannis Niewöhner, Milan Peschel, ...  
13    Klara Castanho, Lucca Picon, Júlia Gomes, Marc...  
18    Freida Pinto, Logan Marshall-Green, Robert Joh...  
...                                                 ...  
1425                                                NaN  
1431  Engin Günaydın, Haluk Bilginer, Binnur Kaya, Ö...  
1436  Juan Sebastián Gutiérrez, Pablo Cano, Pablo Me...  
1466                                                NaN  
1468           Cha In-pyo, Cho Dal-hwan, Song Jae-ryong  

[277 rows x 2 columns]

The entire cast of movies released in 2021 are as listed in the code output above .